contracts/libraries/PermissionsLib.sol

```diff
@@ -31,21 +32,18 @@ library PermissionsLib {
         if (!self.authorized[agent]) {
             self.authorized[agent] = true;
             self.authorizedAgents.push(agent);
+            self.agentToIndex[agent] = self.authorizedAgents.length;
```

Comment:
Shouldn't this be `= self.authorizedAgents.length - 1` ?

Reply:
You are correct. This PR was WIP when I pushed it.